tools/build-rpms.py

```diff
@@ -0,0 +1,254 @@
+                                                    lambda x: x.terminate(), i)]
+        i.wait_until_running()
+        i.reload()
+
+    return instances
+
+
+def setup_ansible(key_material, instances):
+    os.umask(0)
+    keypath = os.path.join(base_dir, "keypair.pem")
+
+    with open(os.open(keypath, os.O_CREAT | os.O_WRONLY, 0o600), 'w') as f:
+        f.write(key_material)
+
+    with open(os.path.join(base_dir, "tools", "appsre-ansible", "inventory"), 'w') as f:
+        f.write("[rpmbuilder]\n")
+        for i in instances:
+            arch = arch_rename_aws.get(i.architecture, i.architecture)
+            f.write(
+                f"{i.public_ip_address} " +
+                ' '.join(
+                    [
+                        f"ansible_ssh_private_key_file={keypath}",
+                        f"ARCH={arch}"]) +
+                "\n")
+
+
+def run_ansible():
+    osbuild_commit = json.load(
+        open(f"{base_dir}/Schutzfile"))["rhel-8.6"]["dependencies"]["osbuild"]["commit"]
+
+    return run_command(["ansible-playbook",
+                        # "-vvv",
+                        "--ssh-extra-args", "-o ControlPersist=no -o StrictHostKeyChecking=no -o ServerAliveInterval=5",
+                        "-i", f"{base_dir}/tools/appsre-ansible/inventory",
+                        f"{base_dir}/tools/appsre-ansible/rpmbuild.yml",
+                        "-e", f"COMPOSER_COMMIT={commit}",
+                        "-e", f"OSBUILD_COMMIT={osbuild_commit}",
+                        "-e", f"RH_ACTIVATION_KEY={os.environ['RH_ACTIVATION_KEY']}",
+                        "-e", f"RH_ORG_ID={os.environ['RH_ORG_ID']}"])
+
+
+def stage_generate_rpms(arch):
+    keyname, key_material = stage("Create keypair", (), create_keypair)
+    instances = stage("Create EC2 instances", (),
+                      create_ec2_instances, arch, keyname)
+    print(instances)
+    stage("Setup ansible", (), setup_ansible, key_material, instances)
+    stage("Run Ansible playbook", (), run_ansible)
+
+
+def check_env():
+    required_envvars = ["RH_ORG_ID", "RH_ACTIVATION_KEY"]
+    if not all((i in os.environ for i in required_envvars)):
+        msg_error(
+            f"At least one of the required environment variables is missing: {required_envvars}")
+        sys.exit(1)
+
+
+def check_params():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--dry-run",
+        help="Only print steps without doing actions",
+        action='store_true')
+
+    parser.add_argument('--base-dir', type=pathlib.Path, required=True,
+                        help='Base directory for creating files')
+
+    parser.add_argument('--commit', type=str, required=True,
+                        help='Commit SHA')
+
+    parser.add_argument('arch', type=str, nargs='+', choices=arch_info.keys(),
+                        help='Architectures to build images for')
+
+    args = parser.parse_args()
+
+    global is_dryrun, base_dir, commit
```

Comment:
I know that passing many variables around is a pain but `global`s... I'm not sure tbh.